### ISSUE TITLE
How to install this leaflet module using npm?

### ISSUE THREAD

**Issue by jcpatac** [calvinmetcalf/leaflet.shapefile#53]
So I want to use this module for my leaflet map but I'm developing it on ionic. How can I use this module?

**Comment by calvinmetcalf** [comment]
copy over the shp.js file leaflet.shapefile.js files into your vendors folder, or really just any folder in your app and then require them (require shp first) this was written way before I know about bundlers or anything

**Comment by jcpatac** [comment]
I did just that but there's an error. I think I'm doing it the wrong way. Any thoughts on this? Thanks!
![image](https://user-images.githubusercontent.com/39621326/41733011-3a5f0fd8-75b5-11e8-9988-6b09b9a6d56e.png)

**Comment by jcpatac** [comment]
So I kind of traced the code and I found out that the ajax.open in shp.js is not working. How do I deal with this? thanks!

**Comment by jcpatac** [comment]
Okay, so I did just that. Inside shp.js, in the binaryAjax function, I also tweaked the ajax.addEventListener because it did not trigger or something like that. So I changed it into ajax.onreadystatechange. The logic/function is still the same but I still got an error. The error is on return resolve(new Buffer(ajax.response));
It seems like the response is null. Any thoughts on this?
![image](https://user-images.githubusercontent.com/39621326/41740843-e6381460-75cb-11e8-90dd-89f9e90b9514.png)

**Comment by calvinmetcalf** [comment]
change `new Buffer(ajax.response)` to `Buffer.from(ajax.response)`

**Comment by calvinmetcalf** [comment]
more generally the [shapefile to geojson library](https://github.com/calvinmetcalf/shapefile-js) is available on npm as `shpjs` and should work, it's only this leaflet plugin that isn't working, you might be better off just using shpjs directly and then using the built in leaflet geojson stuff.

**Comment by jcpatac** [comment]
So I changed `new Buffer(ajax.response)` to `Buffer.from(ajax.response)` but the same error occurred. When I log the response value to the console, it says null. I don't know why though. I think it's because I'm fetching the file locally? I am also in a localhost development server.

**Comment by calvinmetcalf** [comment]
yup local would be the issue, run a server so it's on local host `npm install -g server && serve` will do the trick

**Comment by jcpatac** [comment]
I already did this and the error still occurred. I don't know what went wrong really

**Comment by calvinmetcalf** [comment]
as I said above, this shapefile plugin for leaflet is pretty old, you might be better off using the shapefile to geojson library directly
